homeassistant/components/sensibo/climate.py

```diff
@@ -84,30 +77,40 @@
 HA_TO_SENSIBO = {value: key for key, value in SENSIBO_TO_HA.items()}
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType = None,
+):
     """Set up Sensibo devices."""
-    client = pysensibo.SensiboClient(
-        config[CONF_API_KEY], session=async_get_clientsession(hass), timeout=TIMEOUT
+    _LOGGER.warning(
+        "Loading Sensibo via platform setup is deprecated; Please remove it from your configuration"
     )
-    devices = []
-    try:
-        async with async_timeout.timeout(TIMEOUT):
-            for dev in await client.async_get_devices(_INITIAL_FETCH_FIELDS):
-                if config[CONF_ID] == ALL or dev["id"] in config[CONF_ID]:
-                    devices.append(
-                        SensiboClimate(client, dev, hass.config.units.temperature_unit)
-                    )
-    except (
-        aiohttp.client_exceptions.ClientConnectorError,
-        asyncio.TimeoutError,
-        pysensibo.SensiboError,
-    ) as err:
-        _LOGGER.error("Failed to get devices from Sensibo servers")
-        raise PlatformNotReady from err
-
-    if not devices:
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            SENSIBO_DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Sensibo climate entry."""
+
+    client = hass.data[SENSIBO_DOMAIN][entry.entry_id]["client"]
+
+    devicelist = hass.data[SENSIBO_DOMAIN][entry.entry_id]["devices"]
+    if not devicelist:
         return
 
+    devices = []
```

Comment:
This could be a list comprehension

Reply:
Sorry, don't understand what you mean by this.